x-pack/legacy/plugins/apm/server/lib/index_pattern/getKueryBarIndexPattern.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+import { StaticIndexPattern } from 'ui/index_patterns';
+import { IndexPatternsService } from '../../../../../../../src/legacy/server/index_patterns/service';
+import { Setup } from '../helpers/setup_request';
+
+export const getKueryBarIndexPattern = async ({
+  request,
+  processorEvent,
+  setup
+}: {
+  request: Legacy.Request;
+  processorEvent?: string;
+  setup: Setup;
+}) => {
+  const { config } = setup;
+
+  const indexPatternsService = new IndexPatternsService(
+    (...args: [any, any, any]) =>
+      request.server.plugins.elasticsearch
+        .getCluster('data')
+        .callWithRequest(request, ...args)
+  );
+
+  const indices = processorEvent
+    ? [processorEvent]
+    : ['transaction', 'metric', 'error'];
+
+  const indicesFromConfig = indices.map(index =>
+    index === 'metric'
+      ? config.get<string>(`apm_oss.metricsIndices`)
+      : config.get<string>(`apm_oss.${index}Indices`)
+  );
```

Comment:
There's something that seems a little off. Perhaps it's the implicit mapping between `processorEvent` and config option name. Or perhaps the dynamic resolution of config option name. How will this work with runtime index configurations?

Reply:
Aye, this was a little too implicit. Changed it to:

```typescript
const indicesMap = {
  transaction: indices['apm_oss.transactionIndices'],
  metric: indices['apm_oss.metricsIndices'],
  error: indices['apm_oss.errorIndices']
};

const configuredIndices = indexNames.map(name => indicesMap[name]);
```

and narrowed the type for `processorEvent`.